### DISCUSSION TITLE
Supports to 3DTile Splitting?

### DISCUSSION THREAD

**Discussion by rober72hv** [TerriaJS/terriajs#4960] [Q&A] [answered]
Hi Terria community
I would like to implement the comparison between 3DTiles for Terria next V8.
Specifically I would like that a 3DTile only appear in one area of ​​the map (Left or Right). The same is achieved with the Terrain in the "next" branch.
I'm trying to figure out how TerriaJS manages to split a Terrain, to study how it was implemented. But don't I  see where the magic is done.
I see that the Terria.ts file is added:
`@observable terrainSplitDirection: ImagerySplitDirection =
    ImagerySplitDirection.NONE`

And in the Cesium.ts file:
`this.scene.globe.splitDirection = this.terria.showSplitter
        ? this.terria.terrainSplitDirection
        : ImagerySplitDirection.NONE;`

But from what I've read in the documentation, 3D Tile does not support splitting, only Globe or ImageLayers.
It would help me a lot to know if changes were necessary to Cesium to support the 3D Tile in TerriaJS split tool.
Any help will be greatly appreciated.
Best
Rob

**Comment by moisesrbd** [comment, marked as answer]
Hi Rob
Terria uses a modified version of cesium, called terria-cesium.
The version is placed in the tag similar to the version of cesium, for example: [terria-cesium 1.73.1](https://github.com/TerriaJS/cesium/tree/terriajs-1.73.1) corresponds to version 1.73.1 of cesium.

From what I could see, some changes were made to support dividing a terrain in 2018.
see:
[cesium commit](https://github.com/CesiumGS/cesium/commit/2916a07f23e4eed36be6972378a8de0e94ac7e9c)
Maybe they improved it later.
In the terria-cesium version 1.73.1, many of those changes are present, I hope I can give you an idea.

However the Terrain is very tied to the Globe, through the TerrainProvider, is different with the 3DTiles. I'm not sure it can be done in the same way.

Perhaps using clippingPlanes could help you. It occurs to me that I could create a plane that splits the map, and updates it based on the camera position.

Maybe someone could give more ideas.

I hope it helps you.
Best
Moises

**Comment by KeyboardSounds** [comment]
@moisesrbd's answer is spot on. We don't support 3D Tile splitting yet, but it's highly likely we will be implementing it in the medium-term. Keep an eye on our changelog :) 

Of course, if you urgently want to implement it yourself, I'll help however I can. I've made an issue [here](https://github.com/TerriaJS/terriajs/issues/4965) with some guidance.

Also, as a quick sidenote:
> The version is placed in the tag similar to the version of cesium, for example: terria-cesium 1.73.1 corresponds to version 1.73.1 of cesium.

You are 99% right here, but to nitpick, the patch part of the version string — here the .1 at the end — is _our_ patch version, not Cesium's.
